runtime/executor/src/main/java/org/apache/nemo/runtime/executor/datatransfer/IntermediateDataIOFactory.java

```diff
@@ -60,7 +71,8 @@ public OutputWriter createWriter(final String srcTaskId,
       return new PipeOutputWriter(srcTaskId, runtimeEdge, pipeManagerWorker);
     } else {
       final StageEdge stageEdge = (StageEdge) runtimeEdge;
-      return new BlockOutputWriter(srcTaskId, stageEdge.getDstIRVertex(), runtimeEdge, blockManagerWorker);
+      return new BlockOutputWriter(
+        srcTaskId, stageEdge.getDstIRVertex(), runtimeEdge, blockManagerWorker, memoryManager);
```

Comment:
I think as BlockManagerWorker has MemoryManager, we don't need to pass memoryManager when creating BlockOutputWriter(maybe using `getMemoryManger()`?). If so, we don't need to import MemoryManager here.